src/Workspaces/VisualBasic/Portable/Utilities/SpeculationAnalyzer.vb

```diff
@@ -420,6 +425,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Utilities
             Return False
         End Function
 
+        Private Function ReplacementBreaksWithStatement(originalWithStatement As WithStatementSyntax, replacedWithStatement As WithStatementSyntax) As Boolean
+            Dim originalExpression = originalWithStatement.Expression
+            Dim replacedExpression = replacedWithStatement.Expression
+            Return Not IsReplacementSameType(originalExpression, replacedExpression)
+        End Function
+
+        Private Function IsReplacementSameType(originalExpression As ExpressionSyntax, replacedExpression As ExpressionSyntax) As Boolean
```

Comment:
Consider just inlining this in the previous method since it's only in one place?